Modules/JavaScript/Classes-Constructor-New/homework.js

```diff
@@ -0,0 +1,95 @@
+console.log(`
+  Task.
+
+    Create a Vector object that supports:
+     1. add (same length)
+     2. subtract
+     3. dot
+     4. norm
+     5. If you try to add, subtract, or dot two vectors with different lengths, you must throw an error.
+     6. toString
+     7. equals
+`);
+
+class Vector {
+  list = [];
+
+  constructor(list) {
+    this.list = list;
+  }
+
+  add(vector) {
+    this.checkIfVectorsLengthsMatch(vector);
+    
+    return new Vector(this.list.map((val, index) => val + vector.list[index]));
+  }
+  
+  subtract(vector) {
+    this.checkIfVectorsLengthsMatch(vector);
+    
+    return new Vector(this.list.map((val, index) => val - vector.list[index]));
+  }
+  
+  dot(vector) {
+    this.checkIfVectorsLengthsMatch(vector);
+    
+    return this.list.reduce((sum, val, index) => {
+      console.log('  ----- val=', val, 'vector.list[index]=', vector.list[index], 'addThis=', (val * vector.list[index]))
+      return sum + (val * vector.list[index]);
+    }, 0)
+  }
+  
+  norm() {
+    const sum = this.list.reduce((a, b) => {
+      console.log('  ----- val^2=', Math.pow(b, 2));
+      return a + Math.pow(b, 2);
+    }, 0);
+    return Math.sqrt(sum);
+  }
+
+  toString() {
+    return `(${this.list.toString()})`;
+  }
+
+  equals(vector) {
+    if (this.isLengthEqual(vector)) {
+      for (let i = 0; i < this.list.length; i++) {
+        if (this.list[i] !== vector.list[i]) return false;
+      }
+  
+      return true;
+    }
+
+    return false;
+  }
+
+  isLengthEqual(vector) {
+    if (this.list.length === vector.list.length) {
```

Comment:
Always remember this simple rule, if you are returning true or false from an if statement, it can be simplified.
For example in this case, it could just be return `this.list.length === vector.list.length`. Not a mistake, but do keep it in mind